lib/cloud/file-storage/awsStorage.js

```diff
@@ -0,0 +1,91 @@
+var q = require('q');
+var s3 = require('s3');
+var path = require('path');
+
+var config;
+var awsClient;
+
+function init(storageConfiguration) {
+  config = storageConfiguration;
+  validateConfig();
+  awsClient = s3.createClient(config);
+}
+
+function writeFile(namespace, fileName, fileLocation) {
+  var file;
+  if (namespace) {
+    file = path.join(namespace, fileName);
+  } else {
+    file = fileName;
+  }
+  
+  var params = {
+    localFile: fileLocation,
+    ACL: 'authenticated-read',
+    s3Params: {
+      Bucket: config.bucket,
+      Key: file
+    }
+  };
+  
+  var deferred = q.defer();
```

Comment:
We are migrating to using bluebird vs q.deffered as it is an antipattern https://github.com/petkaantonov/bluebird/wiki/Promise-anti-patterns#the-deferred-anti-pattern